docs/reference/remap/functions/parse_csv.cue

```diff
@@ -0,0 +1,50 @@
+package metadata
+
+remap: functions: parse_csv: {
+	category: "Parse"
+	description: #"""
+		Parses a single CSV formatted row. Only the first row is parsed in case of multiline input value.
+		"""#
+	notices: [
+		"""
+			All values are returned as strings. We recommend manually coercing values to desired types as you see fit.
+			""",
+	]
+
+	arguments: [
+		{
+			name:        "value"
+			description: "The string to parse."
+			required:    true
+			type: ["string"]
+		},
+		{
+			name:        "delimiter"
+			description: "The field delimiter to use when parsing. Must be a single-byte utf8 character."
+			required:    false
+			default: ","
+			type: ["string"]
+		},
+	]
+	internal_failure_reasons: [
+		"delimiter must be a single-byte utf8 character",
+	]
+	return: types: ["array"]
+
+	examples: [
+		{
+			title: "Parse a single CSV formatted row"
+			source: #"""
+				parse_csv!("foo,bar,\"foo \"\", bar\"")
+				"""#
+			return: ["foo", "bar", #"foo ", bar"#]
```

Comment:
I wonder if a follow-up PR could introduce an optional second parameter, to allow you to define the headers of the rows, e.g.:

```rust
parse_csv!("foo,bar", headers: ["header1", "header2"])
// { "header1": "foo", "header2": "bar" }
```

This would then result in an object with the header->field mapping applied.

We don't have to do so here though, but maybe create a tracking issue for this feature, so that someone else can pick this up in the future?

(this would complicate the impl more, as it would have to error or group fields without headers into an "other" key with an array as its value)

Reply:
Good idea - added a feature request #6914